vaultlocker/shell.py

```diff
@@ -69,19 +70,50 @@ def _encrypt_block_device(args, client, config):
     block_uuid = str(uuid.uuid4()) if not args.uuid else args.uuid
     vault_path = _get_vault_path(block_uuid, config)
 
-    dmcrypt.luks_format(key, block_device, block_uuid)
-    # Ensure sym link for new encrypted device is created
-    # LP Bug #1780332
-    dmcrypt.udevadm_rescan(block_device)
-    dmcrypt.udevadm_settle(block_uuid)
-
-    # NOTE: store and validate key
-    client.write(vault_path,
-                 dmcrypt_key=key)
-    stored_data = client.read(vault_path)
-    assert key == stored_data['data']['dmcrypt_key']
+    # NOTE: store and validate key before trying to encrypt disk
+    try:
+        client.write(vault_path,
+                     dmcrypt_key=key)
+    except hvac.exceptions.VaultError as write_error:
+        logger.error(
+            'Vault write to path {}. Failed with error:{}'.format(
```

Comment:
nitpick: it'll look nicer with `error: {}` instead of `error:{}`. Same at similar places below